code/game/objects/items/mail.dm

```diff
@@ -109,9 +109,13 @@
 			playsound(loc, 'sound/machines/twobeep_high.ogg', 100, TRUE)
 
 /obj/item/mail/attack_self(mob/user)
-	if(recipient && user != recipient)
-		to_chat(user, span_notice("You can't open somebody else's mail! That's <em>illegal</em>!"))
-		return
+	if(recipient_ref)
+		var/datum/mind/recipient = recipient_ref.resolve()
+		// If the recipient's mind has gone, then anyone can open their mail
+		// whether a mind can actually be qdel'd is an exercise for the reader
+		if(recipient && recipient != user?.mind)
+			to_chat(user, span_notice("You can't open somebody else's mail! That's <em>illegal</em>!"))
+			return
```

Comment:
Not sure if you'd find it more convenient, but you can do something like weakref == WEAKREF(thing) just fine if you'd like